packages/backend/src/peripherals/zksync/ZksyncClient.ts

```diff
@@ -0,0 +1,112 @@
+import { getErrorMessage, HttpClient, Logger } from '@l2beat/common'
+
+import { parseZksyncResponse } from './parseZksyncResponse'
+import {
+  ZksyncBlocksResultSchema,
+  ZksyncTransactionResultSchema,
+} from './schemas'
+
+export class ZksyncClient {
+  constructor(private httpClient: HttpClient, private logger: Logger) {
+    this.logger = logger.for(this)
+  }
+
+  async getLatestBlock() {
+    const result = await this.call('blocks/lastFinalized')
+
+    const parsed = ZksyncBlocksResultSchema.parse(result)
+
+    return parsed.blockNumber
+  }
+
+  async getTransactionsInBlock(blockNumber: number) {
+    const result = await this.call(`blocks/${blockNumber}/transactions`, {
+      from: 'latest',
+      limit: '100',
+      direction: 'older',
+    })
+
+    const parsed = ZksyncTransactionResultSchema.parse(result)
+
+    if (parsed.list.length === 0) {
+      throw new Error('Transactions list empty!')
+    }
+    let transactions = parsed.list
+
+    let count = parsed.pagination.count
+    while (count > 100) {
```

Comment:
I would actually prefer subtracting the number of txs fetched and comparing against 0